README.md

```diff
@@ -137,82 +137,179 @@ The trace configuration additionally exposes the `samplingRate` option which set
+    * The trace agent is disabled, either because it wasn't started at all, started in disabled mode, or started in an environment where the GCP project ID could not be obtained.
+* `TraceApi#createChildSpan(options)`
+  * `options`: [`TraceOptions`](#trace-span-options)
+  * Returns `?Span`
+  * Attempts to create a child span, and returns a `Span` object if this is successful. Otherwise, it returns `null`. This may be for one of several reasons:
+    * A root span wasn't created beforehand because an earlier call to `runInRootSpan` didn't generate one.
+    * A root span wasn't created beforehand because `runInRootSpan` was not called at all. This likely indicates a programmer error, because child spans should always be nested within a root span.
+    * A root span was created beforehand, but context was lost between then and now. This may also be a programmer error, because child spans should always be created within the context of a root span. See [`Context Propagation`](#context-propagation) for details on properly propagating root span context.
+* `Span#addLabel(key, value)`
+  * `key`: `string`
+  * `value`: `any`
+  * Add a label to the span associated with the calling object. If the value is not a string, it will be stringified with `util.inspect`.
+  * **Note:** Keys and values may be truncated according to the user's configuration and limits set on the Stackdriver Trace API. Keys must be less than 128 bytes, while values must be less than 16 kilobytes, as specified in the [Stackdriver Trace docs][stackdriver-trace-span]. The user may specify a smaller limit on value size through the `maximumLabelValueSize` configuration field.
+* `Span#endSpan()`
+  * Ends the span associated with the calling object. This function should only be called once.
+
+##### Trace Span Options
 
-### Run in span
+Some functions above accept a `TraceOptions` object, which has the following fields:
+
+* `name`: `string`
+  * Required
+  * The name that should be given to the newly created span.
+* `traceContext`: `string`
+  * Optional for root spans, ignored for child spans
+  * A serialized trace context. If the module being traced is a web framework,
+    the plugin that patches this module should attempt to extract this from an
+    incoming request header and set this field; omitting this field may cause
+    trace spans that correspond to a single request across several services in a
+    distributed environment (e.g. microservices) to appear disassociated with
+    one another.
+    See also [Cross-Service Trace Contexts](#cross-service-trace-contexts).
+* `url`: `string`
+  * Optional for root spans, ignored for child spans
+  * The URL of the incoming request. This only applies if the module being
+    traced is a web framework. If given, a label will automatically be created
+    for the new span for the url (under the key `url`). This field will also be
```

Comment:
Our plugins add this label but I don't think the api will do this automatically.